ports/stm32/boards/Passport/modules/tasks/copy_firmware_to_spi_flash_task.py

```diff
@@ -99,8 +106,11 @@ async def copy_firmware_to_spi_flash_task(file_path, size, on_progress, on_done)
                 sf.write(0, buf)  # Need to write the entire page of 256 bytes
 
                 # Success
-                await on_done(None)
+                await on_done(None, None)
+
     except CardMissingError:
-        await on_done(Error.MICROSD_CARD_MISSING)
+        await on_done(Error.MICROSD_CARD_MISSING, None)
+    except Exception as e:
+        await on_done(Error.UNKNOWN, "Error: {}, args: {}".format(type(e), e.args))
```

Comment:
Let's add a more informative error code like `Error.FIRMWARE_UPDATE_FAILED`.

And let's change the message string to the following to have a cleaner exception name and to show the args/message cleaner in the 1 arg case:

`await on_done(Error.UNKNOWN,"Error: {}, Info: {}".format(e.__class__.__name__, e.args[0] if len(e.args) == 1 else e.args))`